File: pens/pens.py
#Scrivi qui il tuo codice
#Scrivi qui il tuo codice
N = int(input().strip())
P = [None] * N
P = list(map(int, input().strip().split()))
#print(P)
count = N
for i in range(N):
    if P[i] == 0:
        #print(P[min(i + 1, N - 1)], min(i + 1, N - 1))
        if P[max(i - 1, 0)] > 1:
            count -= 1
            P[max(i - 1, 0)] -= 1
        elif P[min(i + 1, N - 1)] > 1:
            count -= 1
            P[min(i + 1, N - 1)] -= 1
        #print(P)
    else:
        count -= 1
        #print(i)

print(count)

File: pens/pens.cpp
#include <bits/stdc++.h>

using namespace std;

int main(){
	
//	freopen("input.txt", "r", stdin);
//	freopen("output.txt", "w", stdout);
	
	int N;
	
	cin >> N;
	
	vector<int> P(N);
	
	for (int i = 0; i < N; i++){
		cin >> P[i];
	}
	
	int count = N;
	
	for (int i = 0; i < N; i++){
		if (P[i] == 0){
			if (P[max(i - 1, 0)] > 1){
				count--;
				P[max(i - 1, 0)]--;
			} else if (P[min(i + 1, N - 1)] > 1){
				count--;
				P[min(i + 1, N - 1)]--;
			}
		} else {
			count--;
		}
		
	}
	
	cout << count << "\n";
	
	return 0;
}
